pkgs/top-level/all-packages.nix

```diff
@@ -20191,7 +20191,9 @@ with pkgs;
 
   libcint = callPackage ../development/libraries/libcint { };
 
-  libclc = callPackage ../development/libraries/libclc { };
+  libclc = callPackage ../development/libraries/libclc {
+    llvmPackages = llvmPackages_latest;
+  };
```

Comment:
I just build `libclc` successfully against the regular `llvmPackages` on NixOS 22.11.

<details>

<summary>Derivation</summary>

```json
{
  "/nix/store/l4ccy0bjw1h8i8j32w4xrxrbijxy109r-libclc-14.0.6.drv": {
    "outputs": {
      "out": {
        "path": "/nix/store/sws900kbpmmjwpfsfdp3kn9x7mgpmw1c-libclc-14.0.6"
      }
    },
    "inputSrcs": [
      "/nix/store/9krlzvny65gdc8s7kpb6lkx8cd02c25b-default-builder.sh"
    ],
    "inputDrvs": {
      "/nix/store/05n8dblmwlkafylbai3hhx4nz88g2vbz-clang-11.1.0.drv": [
        "dev",
        "out"
      ],
      "/nix/store/11dz31m8rb8lmh9r9jy6imph8wbqmsv3-cmake-3.24.3.drv": [
        "out"
      ],
      "/nix/store/1kbciw3fk38lvwv5kmc4zgz0f6dqldx8-source.drv": [
        "out"
      ],
      "/nix/store/2knhp55njxnhh7j1l0w7ns0hi6bxmn1k-stdenv-linux.drv": [
        "out"
      ],
      "/nix/store/3sjd4vvbdssljiy45a9igjy9kkxd2skq-bash-5.1-p16.drv": [
        "out"
      ],
      "/nix/store/8gka1ln4n7xi55hgkj8qdixydg7kwlcf-ninja-1.11.1.drv": [
        "out"
      ],
      "/nix/store/hgv33pjyznisg95ljrygyxk4xc7fbrp5-llvm-11.1.0.drv": [
        "dev"
      ],
      "/nix/store/kxfzn11kf6nh0igzsgfm72zfxj77lsda-python3-3.10.8.drv": [
        "out"
      ],
      "/nix/store/lrljsy25g5las83hhfxlqrjgch4dlm53-SPIRV-LLVM-Translator-unstable-2022-05-04.drv": [
        "out"
      ]
    },
    "system": "x86_64-linux",
    "builder": "/nix/store/dsd5gz46hdbdk2rfdimqddhq6m8m8fqs-bash-5.1-p16/bin/bash",
    "args": [
      "-e",
      "/nix/store/9krlzvny65gdc8s7kpb6lkx8cd02c25b-default-builder.sh"
    ],
    "env": {
      "buildInputs": "/nix/store/kzf7caq6grrh0gjchwajv9v264vfwlr9-llvm-11.1.0-dev /nix/store/0ammdv24jk6iz35gd7psha0li6g6k5d4-clang-11.1.0-dev",
      "builder": "/nix/store/dsd5gz46hdbdk2rfdimqddhq6m8m8fqs-bash-5.1-p16/bin/bash",
      "cmakeFlags": "-DCMAKE_INSTALL_INCLUDEDIR=include",
      "configureFlags": "",
      "depsBuildBuild": "",
      "depsBuildBuildPropagated": "",
      "depsBuildTarget": "",
      "depsBuildTargetPropagated": "",
      "depsHostHost": "",
      "depsHostHostPropagated": "",
      "depsTargetTarget": "",
      "depsTargetTargetPropagated": "",
      "doCheck": "",
      "doInstallCheck": "",
      "mesonFlags": "",
      "name": "libclc-14.0.6",
      "nativeBuildInputs": "/nix/store/rxb9zl1mdx9f2n3z0nyv9zhngj3f2pw0-cmake-3.24.3 /nix/store/lfj3vna0mcps3fcjlg5gw4jss05kvgry-ninja-1.11.1 /nix/store/zdba9frlxj2ba8ca095win3nphsiiqhb-python3-3.10.8 /nix/store/91i4dkyivd1i5jljyy2l14pbbyf8kwn1-SPIRV-LLVM-Translator-unstable-2022-05-04",
      "out": "/nix/store/sws900kbpmmjwpfsfdp3kn9x7mgpmw1c-libclc-14.0.6",
      "outputs": "out",
      "patches": "",
      "pname": "libclc",
      "postPatch": "substituteInPlace CMakeLists.txt \\\n  --replace 'find_program( LLVM_CLANG clang PATHS ${LLVM_BINDIR} NO_DEFAULT_PATH )' \\\n            'find_program( LLVM_CLANG clang PATHS \"/nix/store/6zka9igv5cif9k4h6zf37y9sjlnbq8n3-clang-11.1.0/bin\" NO_DEFAULT_PATH )' \\\n  --replace 'find_program( LLVM_SPIRV llvm-spirv PATHS ${LLVM_BINDIR} NO_DEFAULT_PATH )' \\\n            'find_program( LLVM_SPIRV llvm-spirv PATHS \"/nix/store/91i4dkyivd1i5jljyy2l14pbbyf8kwn1-SPIRV-LLVM-Translator-unstable-2022-05-04/bin\" NO_DEFAULT_PATH )'\n",
      "propagatedBuildInputs": "",
      "propagatedNativeBuildInputs": "",
      "sourceRoot": "source/libclc",
      "src": "/nix/store/v4nsp8wi4p9iv73202k9kp0hbcj1lfcr-source",
      "stdenv": "/nix/store/kmfaajdpyyyg319vfqni5jm9wkxjmf73-stdenv-linux",
      "strictDeps": "1",
      "system": "x86_64-linux",
      "version": "14.0.6"
    }
  }
}
```

</details>

I've also push the path to my [Cachix](https://app.cachix.org/cache/hmenke).